src/widgets/menu/lv_menu.c

```diff
@@ -142,7 +142,13 @@ lv_obj_t * lv_menu_page_create(lv_obj_t * menu, char const * const title)
 
 lv_obj_t * lv_menu_cont_create(lv_obj_t * parent)
 {
-    LV_ASSERT_OBJ(parent, &lv_menu_page_class);
+    LV_ASSERT_NULL(parent);
+    if(!parent || (LV_USE_ASSERT_OBJ &&
+                   !(lv_obj_has_class(parent->class_p, &lv_menu_page_class)
+                     || lv_obj_has_class(parent->class_p, &lv_menu_section_class)))) {
```

Comment:
I have accidentally led you astray

```suggestion
                   !(lv_obj_has_class(parent, &lv_menu_page_class)
                     || lv_obj_has_class(parent, &lv_menu_section_class)))) {
```